### PR TITLE
DDL schema locking

### PR DESCRIPTION
**Pull request by akshatsikarwar**
Should address issue #286

### CODE REVIEW

**Comment by nirbhayc** [comment]
Looks good to me. I'd suggest not to squash the commits while merging.

**Comment by akshatsikarwar** [comment]
holy moly

File: db/osqlblockproc.c [hunk 1]
```diff
@@ -1279,17 +1280,14 @@ static int apply_changes(struct ireq *iq, blocksql_tran_t *tran, void *iq_tran,
     listc_init(&iq->bpfunc_lst, offsetof(bpfunc_lstnode_t, linkct));
 
     /* go through the complete list and apply all the changes */
+    if (iq->tranddl) pthread_mutex_lock(&ddl_lk);
```
Comment (ldennis) [inline db/osqlblockproc.c]: Can we have an option to enable nonblocking on this with trylock? IIRC, we used to return "schema change in progress" error.

Comment (akshatsikarwar) [inline db/osqlblockproc.c]: Sure - that's a good idea.

**Comment by akshatsikarwar** [comment]
Failure in `reqlog` accessing free'd memory looks legit. `pglogs_asof` doesn't look related.

**Comment by ldennis** [comment]
Looks good

**Comment by ldennis** [comment]
With this patch, I wonder if the broken sc seed is still worth fixing.

**Comment by akshatsikarwar** [comment]
why does `sc_resume` keep failing?

**Comment by riverszhang89** [comment]
Not sure. master is failing on that test as well.

**Pull request merged**